src/addresses.rs

```diff
@@ -95,18 +146,26 @@ pub fn find_address(
             poi,
             geofinder.get(&poi.coord),
         )),
-        _ => rubber
-            .get_address(&poi.coord)
-            .map_err(|e| {
-                warn!("get_address returned ES error for {}: {}", poi.id, e);
-                e
-            })
-            .ok()
-            .and_then(|addrs| addrs.into_iter().next())
-            .map(|addr| {
-                addr.address()
-                    .expect("get_address returned a non-address object")
-            }),
+        _ => {
+            if let Some(addr) = get_current_addr(rubber, poi_dataset, &poi.id) {
+                // Use the address already in ES
+                Some(addr)
```

Comment:
a few more checks will be necessary here:
 * has the poi position not changed ?
 * has this addr not been read from tags (that are deleted now) ?